owncloudApp/src/main/java/com/owncloud/android/presentation/logging/LogsListActivity.kt

```diff
@@ -108,4 +120,46 @@ class LogsListActivity : AppCompatActivity() {
 
         recyclerViewLogsAdapter.setData(items)
     }
+
+    private fun uploadFile(file: File) {
+        val destinationPath = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath
+        val destinationFolder = File(destinationPath)
+        val originalName = file.name
+        var uniqueName = originalName
+        var fileNumber = 1
+
+        while (File(destinationFolder, uniqueName).exists()) {
+            val fileNameParts = originalName.split(".")
+            uniqueName = "${fileNameParts[0]}_${fileNumber}.${fileNameParts[1]}.log"
+            fileNumber++
+        }
```

Comment:
I'm not sure about this algorithm. Did you think about the case when the name of the app (the first part of the log file name) contains a `.`? For example, `my.cloud`. The name of the log file would be malformed.
Also, instead of `name_X.YYYY-MM-DD_HH:mm:SS.log` I would bet for `name.YYYY-MM-DD_HH:mm:SS (X).log`, which is way more standard

Reply:
done